utils/data_retrieval.py

```diff
@@ -25,6 +26,59 @@ def trim_base_from_uri(uri_to_trim):
     return uri_to_trim.replace(TILED_BASE_URI, "")
 
 
+def get_processed_experiment_names():
+    try:
+        client = from_uri(TILED_URI, api_key=TILED_API_KEY)
+        processed_list_of_experiments = client["processed"].keys()
+        list_of_experiments = [
+            experiment for experiment in processed_list_of_experiments
+        ]
+        return list_of_experiments
+    except Exception as e:
+        error_message = f"Error retrieving experiment names: {e}"
+        print(error_message)
+        return []
+
+
+def get_csv_file_uri(experiment_name):
+    try:
+        client = from_uri(TILED_URI, api_key=TILED_API_KEY)
+        processed_experiment_uri = client["processed"][experiment_name].uri
+        get_csv_file_uri = processed_experiment_uri + f"/{experiment_name}"
+        return get_csv_file_uri
+    except Exception as e:
+        error_message = f"Error retrieving csv file uri: {e}"
+        print(error_message)
+        return None
+
+
+def tiled_read_csv(csv_file_uri):
+    try:
+        client = from_uri(csv_file_uri, api_key=TILED_API_KEY)
+        csv_data = client.read()
+        return csv_data
+    except Exception as e:
+        error_message = f"Error retrieving csv file: {e}"
+        print(error_message)
+        return None
+
+
+def write_csv_from_interface(experiment_name, data):
+    try:
+        # TODO: optimize this with a get_csv_file_uri function later
+        client = from_uri(TILED_URI, api_key=TILED_API_KEY, include_data_sources=True)
+        container_client = client["processed"][experiment_name]
+        csv_client = container_client[experiment_name]
```

Comment:
When not iterating, we can just access the final node directly by 
`csv_client = from_uri([TILED_BASE_URI, "processed", experiment_name, experiment_name].join("/"), ...)`